virtual-programs/shapes.folk

```diff
@@ -1,111 +1,301 @@
-# sides 2 => line
-# sides 3 => triangle
-# sides 4 => square
+set shapes [dict create triangle 3 square 4 pentagon 5 hexagon 6 septagon 7 octagon 8 nonagon 9]
+
+proc process_offset {offset region} {
```

Comment:
I wonder if 1. this is duplicating stuff we already do with the `region` operators and 2. we should have something like `region` but `point` and operate on the center point instead, so it's like `Wish $this draws a triangle with center [point move [center $this] 50%]`... idk. This is probably ok for now. It's just weird that all the logic is tightly bound and sealed inside shapes.folk when this is a capability we could imagine using elsewhere

Reply:
Oh yeah this is a good point, and idk if the API would be exactly that but I see what you're getting at with manipulating the center point rather than operating on the region. Definitely worth putting something like this in the global namespace to avoid duplicating!